src/restSamples.ts

```diff
@@ -0,0 +1,222 @@
+                // the user canceled the selection
+                if (!selection) return;
+
+                let swagger = require(selection['file']);
+                let swaggerHandler = new Swagger(swagger, selection['file'].split('/').slice(0, -1).join('/'));
+                if (selection['example'] == "From Definition") {
+                    let playbook = swaggerHandler.generateRestApiTasks(selection['path'], selection['method'], null);
+                    pm.insertTask(playbook);
+                } else {
+                    let playbook = swaggerHandler.generateRestApiTasks(selection['path'], selection['method'],  selection['example']);
+                    pm.insertTask(playbook);
+                }
+            });
+        });
+    }
+
+    public getSpecificationLocation(cb) {
+        let config = vscode.workspace.getConfiguration('ansible');
+
+        if (false /*config.has('AzureRestAPISpecificationPath') && config.get('AzureRestAPISpecificationPath') != ""*/) {
+            cb(config.get('AzureRestAPISpecificationPath'));
+        } else {
+
+            this._outputChannel.append('\nConnecting to Cloud Shell.');
+            this._outputChannel.show();
+            const progress = utilities.delayedInterval(() => { this._outputChannel.append('.') }, 500);
+
+            this._outputChannel.appendLine("Getting Azure REST API specifications...");
+            let clone = require('git-clone');
+            let home: string = path.join(process.env[(process.platform == 'win32') ? 'USERPROFILE' : 'HOME'], 'azure-rest-api-specs');
+            clone("https://github.com/Azure/azure-rest-api-specs.git", home, null, (result) => {
+
+                config.update('AzureRestAPISpecificationPath', home, vscode.ConfigurationTarget.Global);
+                if (result == undefined) {
+                    this._outputChannel.appendLine("REST API feature ready...");
+                } else {
+                    this._outputChannel.appendLine("Failed to acquire REST API specifications...");
+                }
+                progress.cancel();
+                cb(config.get('AzureRestAPISpecificationPath'))
+            })
+        }            
+    }
+
+    private queryAll(path: string) {
+        let operations = {};
+        let __this = this;
+
+        __this.queryApiGroup(path, function (dirs: string[]) {
+            dirs.forEach(dir => {
+                __this.queryDirectory(dir, true, ".json", function(files) {
+                    if (files != null) {
+                        files.forEach(file => {
+                            let swagger = require(dir + '/' + file);
+                            for (var path in swagger.paths) {
+                                for (var method in swagger.paths[path]) {
+                                    // add only if there are examples
+                                    if (swagger.paths[path][method]['x-ms-examples'] != undefined) {
```

Comment:
!swagger.paths[path][method]['x-ms-examples'] , to cover both undefined, null and empty check